plasma_framework/contracts/src/vaults/predicates/IErc20DepositVerifier.sol

```diff
@@ -0,0 +1,13 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import {PaymentTransactionModel as DepositTx} from "../../transactions/PaymentTransactionModel.sol";
+
+contract IErc20DepositVerifier {
```

Comment:
This is an abstract contract instead of an interface because:
- If it was an interface, then the `verify` method would need to be declared `external`
- and that would mean that the `_depositTx` param would need to be declared `calldata`
- ... but this causes an `UnimplementedFeatureError` from the solidity compiler.

There's no more info, but I imagine it has to do with `pragma experimental ABIEncoderV2`

Reply:
yes, it seems like struct + public would crash if the caller does not state "pragma experimental ABIEncoderV2".
And sadly there would not be warning error that point you where to fix :(
https://github.com/ethereum/solidity/issues/5659